modules/imgcodecs/src/grfmt_jpeg2000_openjpeg.cpp

```diff
@@ -37,49 +39,233 @@ const String colorspaceName(COLOR_SPACE colorspace)
 }
 
 
-template<typename OutT, bool doShift, typename InT, std::size_t N>
-void copy_data_(Mat& out, const std::array<InT*, N>& data, uint8_t shift) {
+template<typename OutT, typename InT>
```

Comment:
doShift is a template parameter for a reason ... do not evaluate it for every pixel

Reply:
Actually it is evaluated once per image, basically it is possible to split functions to not evaluate `isShiftRequered` every *row*, but for continueus mat it is evaluated once at all